runtime/compiler/control/J9Options.cpp

```diff
@@ -2456,15 +2456,14 @@ bool J9::Options::feLatePostProcess(void * base, TR::OptionSet * optionSet)
       self()->setOption(TR_DisableIntrinsics);
       }
 
-   /* Temporary (until SVM is the default)
-    *
-    * If the SVM is not enabled, use the old _coldUpgradeSampleThreshold
+   /* If the SVM is enabled, use a higher _coldUpgradeSampleThreshold
     * for AGGRESSIVE_AOT
     */
-   if (!self()->getOption(TR_EnableSymbolValidationManager)
-       && TR::Options::_aggressivenessLevel == TR::Options::AGGRESSIVE_AOT)
+   if (TR::Options::_aggressivenessLevel == TR::Options::AGGRESSIVE_AOT &&
+       self()->getOption(TR_EnableSymbolValidationManager) &&
+       TR::Options::sharedClassCache())
       {
-      TR::Options::_coldUpgradeSampleThreshold = 10;
+      TR::Options::_coldUpgradeSampleThreshold = 20;
```

Comment:
Hm...I guess we need to take into account the fact that the user could set this differently.

Reply:
Order of options processing:

JIT:
```
JIT_INITIALIZED
    onLoadInternal
        processOptionsAOT/processOptionsJIT
            fePreProcess
            jitPreProcess
            processOptions
```
SCC:
```
JIT_INITIALIZED
    if ((rc = j9shr_init(vm, loadFlags, &nonfatal)) != J9VMDLLMAIN_OK)
        vm->sharedClassConfig->runtimeFlags |= J9SHR_RUNTIMEFLAG_CACHE_INITIALIZATION_COMPLETE;
```
            
JIT:
```
ABOUT_TO_BOOTSTRAP
    aboutToBootstrap
        latePostProcessAOT/latePostProcessJIT
            latePostProcess
                feLatePostProcess
                    TR::Options::_coldUpgradeSampleThreshold = 20;
```
Basically:
`JIT_INITIALIZED`: preset options values and process user set options, but we don't know if we can AOT
`ABOUT TO BOOTSTRAP`: we know we can AOT, but we don't know if user specified options

Heh, @mpirvu could we just bite the bullet that under `-Xtune:virtualized` the user can't set `-Xjit:coldUpgradeSampleThreshold`?

Reply:
I guess I could add an env var to prevent the override of this... Do you think it's worth it?

Reply:
There is already this piece of code that tries to modify the threshold when AOT is used with SCC:
```
            // For AOT with GCR enabled we can be more conservative with
            // upgrades through sampling because we can rely on GCR
            // In this case we'll use the same threshold value as the one
            // employed in `setGlobalAggressiveAOT` (note that larger values
            // means more conservative upgrades and that the default value is
            // currently TR_DEFAULT_COLD_UPGRADE_SAMPLE_THRESHOLD==3)
            if (!self()->getOption(TR_DisableGuardedCountingRecompilations) &&
              _coldUpgradeSampleThreshold == TR_DEFAULT_COLD_UPGRADE_SAMPLE_THRESHOLD)
               _coldUpgradeSampleThreshold = 10;
```
You could do something very similar for your case, just test the Xtune:option. Of course you may want to test whether the user has set some other value by checking against the value set by Xtune (I would like something symbolic in this case, like TR_XTUNE_COLD_UPGRADE_SAMPLE_THRESHOLD)
This would be an OMR change.

Reply:
Yeah looks reasonable, though I guess it would still not work if the user set the value to be the default set in `preprocess` (ie 10, since we wouldn't be able to tell if it was preset to 10 or user-set to 10; in the latter case we ideally shouldn't override it to 20)

> This would be an OMR change.

:sob: